code_conventions.md

```diff
@@ -172,156 +125,46 @@ for (int i = 0; i < len; i++)
-----
-Strings are double quotes. While python and php allow single and double quoted strings, and PEP8 only says "pick a rule and stick to it". We decided to do double quotes to match C++.
+Go from public, protected to private. When using a class, one is usually more interested in the public items. For inheriting the protected ones can be interesting, while the private parts should only be meaningful to the maintainer of the class itself.
 
-Lines and Linelength
-----
-* "Maximum line length" - There is no hard limit on the line length, but as a thumb of rule, try to keep it to at most 120 characters. This helps doing the code reviews!
-* A single line contains only one statement
+## Strings
+Strings MUST be defined by using double quotes.
 
-Alignment
-----
-Align code for better readability.
-This can be done on assignment level, parameter passing on function, array definitions and so on. 
+## Lines and length
+There is no hard limit on the line length, but as a thumb of rule, try to keep it to at most 120 characters.
 
+A single line SHOULD only contain one statement.
 
-Code Guidelines
-====
-Below are a couple of guidelines which should be followed, unless there's good reason not to.
+## Class Files
+Each class SHOULD have its own file with a filename corresponding to the class name.
 
-Implementation (C/C++)
-----
-All implementations should be in .cpp files. An exception are template functions, which must be implemented in the header file.
+Hence a class named Printer would have a printer.h + printer.cpp for C/C++, a printer.py for Python and a printer.php for PHP.
 
-Sometimes including the implementation in the header file can make it easier for the compiler to inline functions.
+Every class SHOULD be in it's own file, unless it's a private class.
 
-Class Files
-----
-Each class would have its own file with a filename corrseponding to the class name.
-Hence a class named Printer would have a printer.h + printer.cpp for C/C++, a printer.py for Python and a printer.php for PHP
-
-Namespaces
-----
+## Namespaces
 [ TO BE DISCUSSED ]
 
-Const vs Non-const (C/C++)
-----
-The best practice is to use const when and wherever possible. This is for both arguments declared in functions as well as the return values of the functions and the functions themselves (when they don't change the internal state of the object).
-In the long run this will make the code, libraries and runtime more stable and robust.
-In the short run this might cause some friction with (older) code that does not use this concept (yet).
-
-Pointers vs. References used as return values in argument list
------
-[ TO BE DISCUSSED ]
-If arguments are allowed to be used to return values, then for all arguments everywhere, using doxygen commenting, [in] and [out] tags must be put in the comments describing the arguments.
+## Functions
+* If a function needs to return more than one value, the return value could be a dictionary/hashtable construction or a specially defined struct.
```

Comment:
Erhm, shouldn't this be left to what the language supports?

Reply:
Hence, could. Not must or should.